packages/system/src/MasterList/ListView/api.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  MasterListsQuery,
+  MasterListSortFieldInput,
+  OmSupplyApi,
+  SortBy,
+  ListApi,
+  FilterBy,
+} from '@openmsupply-client/common';
+import { MasterList, MasterListRow } from '../types';
+
+const masterListsGuard = (masterListsQuery: MasterListsQuery) => {
+  if (masterListsQuery.masterLists.__typename === 'MasterListConnector') {
+    return masterListsQuery.masterLists;
+  }
+
+  if (masterListsQuery.masterLists.__typename === 'ConnectorError') {
+    throw new Error(masterListsQuery.masterLists.error.description);
+  }
+
+  throw new Error('oh dear');
```

Comment:
The `__typename` field is not optional, so this isn't needed. Sometimes it's optional and I can't really determine why that is, but when it is optional, typescript complains and this throw is required.

Reply:
thanks - I was trying to find why it was sometimes optional; the error conditions are a mix of required and optional and that bugged me 🤷 
For this one - ts was complaining.. but now is fine?? have removed it.